lib/open_api/serializer.rb

```diff
@@ -1,14 +1,42 @@
 module OpenApi
   module Serializer
     def as_json(arg = {})
-      return super unless kind_of?(ApplicationRecord)
+      previous = super
+      encrypted_columns_set = (self.class.try(:encrypted_columns) || []).to_set
+      encryption_filtered = previous.except(*encrypted_columns_set)
+      version = version_from_path(arg)
 
-      super.tap do |hash|
-        hash.each_key do |attr_key|
-          hash[attr_key] = hash[attr_key].iso8601 if hash[attr_key].kind_of?(Time)
-          hash[attr_key] = hash[attr_key].to_s if attr_key.ends_with?("_id") || attr_key == "id"
-        end
+      return encryption_filtered unless arg.key?(:prefixes) || version
+
+      schema = Api::Docs[version].definitions[self.class.name]
+
+      filter(encryption_filtered.slice(*schema["properties"].keys), schema, encrypted_columns_set)
+    end
+
+    private
+
+    def version_from_path(arg)
+      url = ManageIQ::API::Common::Request.current.instance_variable_get(:@original_url) || nil
+      base_path = url ? URI.parse(url).path : arg[:prefixes].first
+      api_version_from_prefix(base_path)
+    end
+
+    def filter(attrs, schema, encrypted_columns_set)
+      schema["properties"].keys.each do |name|
+        next if attrs[name].nil?
+
+        attrs[name] = attrs[name].iso8601 if attrs[name].kind_of?(Time)
+        attrs[name] = attrs[name].to_s if name.ends_with?("_id") || name == "id"
+        attrs[name] = public_send(name) if !attrs.key?(name) && !encrypted_columns_set.include?(name)
       end
+      attrs.compact
+    end
+
```

Comment:
@syncrou Are the attribute name and column names always a one to one match between the openapi.json and the models or do the need to be camel cased or snake cased?

Reply:
They're 'assumed' to be the same currently. I should add specs to validate they match.  There actually may tests already checking the attribute names in common.

Reply:
@mkanoor I added the requested specs.